mass-deployer/pkg/mass-deployer/deployer.go

```diff
@@ -0,0 +1,206 @@
+	filter := types.NodeFilter{}
+	statusUp := "up"
+	filter.Status = &statusUp
+	filter.TotalCRU = &group.FreeCPU
+	filter.FreeMRU = &group.FreeMRU
+
+	if group.FreeSSD > 0 {
+		ssd := uint64(convertGBToBytes(int(group.FreeSSD)))
+		filter.FreeSRU = &ssd
+	}
+	if group.FreeHDD > 0 {
+		hdd := uint64(convertGBToBytes(int(group.FreeHDD)))
+		filter.FreeHRU = &hdd
+	}
+	if group.Regions != "" {
+		filter.Region = &group.Regions
+	}
+	if group.Certified {
+		certified := "Certified"
+		filter.CertificationType = &certified
+	}
+	if group.Pubip4 {
+		filter.IPv4 = &group.Pubip4
+	}
+	if group.Pubip6 {
+		filter.IPv6 = &group.Pubip6
+	}
+	if group.Dedicated {
+		filter.Dedicated = &group.Dedicated
+	}
+
+	freeSSD := []uint64{group.FreeSSD}
+	if group.FreeSSD == 0 {
+		freeSSD = nil
+	}
+	freeHDD := []uint64{group.FreeHDD}
+	if group.FreeSSD == 0 {
+		freeHDD = nil
+	}
+
+	nodes, err := deployer.FilterNodes(ctx, d.TFPluginClient, filter, freeSSD, freeHDD, nil, int(group.NodesCount))
+	return nodes, err
+}
+
+func (d Deployer) ParseVms(vms []parser.Vm, groups map[string][]int, sshKeys map[string]string) (map[string][]workloads.VM, map[string][][]workloads.Disk) {
+	vmsWorkloads := map[string][]workloads.VM{}
+	vmsDisks := map[string][][]workloads.Disk{}
+	for _, vm := range vms {
+		// make sure the group is valid
+		if _, ok := groups[vm.Nodegroup]; !ok {
+			continue
+		}
+		sshKey := sshKeys[vm.SSHKey]
+
+		w := workloads.VM{
+			Flist:      vm.Flist,
+			CPU:        vm.FreeCPU,
+			Memory:     vm.FreeMRU,
+			PublicIP:   vm.Pubip4,
+			PublicIP6:  vm.Pubip6,
+			Planetary:  vm.Planetary,
+			RootfsSize: convertGBToBytes(vm.Rootsize),
+			Entrypoint: vm.Entrypoint,
+			EnvVars:    map[string]string{"SSH_KEY": sshKey},
+		}
+
+		var disks []workloads.Disk
+		var mounts []workloads.Mount
+		for _, disk := range vm.SSDDisks {
+			DiskWorkload := workloads.Disk{
+				Name:   fmt.Sprintf("%sdisk", vm.Name),
+				SizeGB: convertGBToBytes(disk.Capacity),
+			}
+
+			disks = append(disks, DiskWorkload)
+			mounts = append(mounts, workloads.Mount{DiskName: DiskWorkload.Name, MountPoint: disk.Mount})
+		}
+		w.Mounts = mounts
+
+		for i := 0; i < vm.Count; i++ {
+			w.Name = fmt.Sprintf("%s%d", vm.Name, i)
+			vmsWorkloads[vm.Nodegroup] = append(vmsWorkloads[vm.Nodegroup], w)
+			vmsDisks[vm.Nodegroup] = append(vmsDisks[vm.Nodegroup], disks)
+		}
+	}
+	return vmsWorkloads, vmsDisks
+}
+
+func (d Deployer) MassDeploy(ctx context.Context, vms []workloads.VM, nodes []int, disks [][]workloads.Disk) ([]VMInfo, error) {
+	networks := make([]*workloads.ZNet, len(vms))
+	vmDeployments := make([]*workloads.Deployment, len(vms))
+
+	var lock sync.Mutex
+	var wg sync.WaitGroup
+
+	nodesCount := len(nodes)
+	deploymentInfo := []vmDeploymentInfo{}
+
+	for i, vm := range vms {
+		nodeID := nodes[i%nodesCount]
+		wg.Add(1)
+
+		go func(vm workloads.VM, i int, nodeID uint32) {
+			defer wg.Done()
+
+			network := workloads.ZNet{
+				Name:        generateRandomString(10),
```

Comment:
we can use the vm name and add network to it

Reply:
the vm name is used for multible vms so having network to be network+vm name would get conflict error